operator/internal/central/carotation/rotation_test.go

```diff
@@ -0,0 +1,206 @@
+			},
+		},
+		{
+			name:   "delete secondary CA",
+			action: DeleteSecondary,
+			additionalSetup: func(t *testing.T, old types.SecretDataMap) {
+				secondary, err := certgen.GenerateCA()
+				require.NoError(t, err)
+				certgen.AddSecondaryCAToFileMap(old, secondary)
+			},
+			assert: func(t *testing.T, old, new types.SecretDataMap) {
+				require.Equal(t, old[mtls.CACertFileName], new[mtls.CACertFileName], "primary CA cert should be unchanged")
+				require.Equal(t, old[mtls.CAKeyFileName], new[mtls.CAKeyFileName], "primary CA key should be unchanged")
+				require.NotContains(t, new, mtls.SecondaryCACertFileName, "secondary CA cert should be removed")
+				require.NotContains(t, new, mtls.SecondaryCAKeyFileName, "secondary CA key should be removed")
+			},
+		},
+		{
+			name:   "no rotation action, secondary CA not present",
+			action: NoAction,
+			assert: func(t *testing.T, old, new types.SecretDataMap) {
+				require.Equal(t, old[mtls.CACertFileName], new[mtls.CACertFileName], "primary CA cert should be unchanged")
+				require.Equal(t, old[mtls.CAKeyFileName], new[mtls.CAKeyFileName], "primary CA key should be unchanged")
+				require.NotContains(t, new, mtls.SecondaryCACertFileName, "secondary CA cert should be present")
+				require.NotContains(t, new, mtls.SecondaryCAKeyFileName, "secondary CA key should be present")
+			},
+		},
+		{
+			name:   "no rotation action, secondary CA present",
+			action: NoAction,
+			additionalSetup: func(t *testing.T, old types.SecretDataMap) {
+				secondary, err := certgen.GenerateCA()
+				require.NoError(t, err)
+				certgen.AddSecondaryCAToFileMap(old, secondary)
+			},
+			assert: func(t *testing.T, old, new types.SecretDataMap) {
+				require.Equal(t, old[mtls.CACertFileName], new[mtls.CACertFileName], "primary CA cert should be unchanged")
+				require.Equal(t, old[mtls.CAKeyFileName], new[mtls.CAKeyFileName], "primary CA key should be unchanged")
+				require.Equal(t, old[mtls.SecondaryCACertFileName], new[mtls.SecondaryCACertFileName], "secondary CA cert should be unchanged")
+				require.Equal(t, old[mtls.SecondaryCAKeyFileName], new[mtls.SecondaryCAKeyFileName], "secondary CA key should be unchanged")
+			},
+		},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+
+			primary, err := certgen.GenerateCA()
+			require.NoError(t, err)
+			fileMap := make(types.SecretDataMap)
+			certgen.AddCAToFileMap(fileMap, primary)
+			err = certgen.IssueCentralCert(fileMap, primary, mtls.WithNamespace("stackrox"))
+			require.NoError(t, err)
+
+			if tt.additionalSetup != nil {
+				tt.additionalSetup(t, fileMap)
+			}
+
+			oldFileMap := maputil.ShallowClone(fileMap)
+			err = Handle(tt.action, fileMap)
+			require.NoError(t, err)
+
+			tt.assert(t, oldFileMap, fileMap)
+		})
+	}
+}
+
+func generateTestCertWithValidity(t *testing.T, notBeforeStr, notAfterStr string) *x509.Certificate {
+	t.Helper()
```

Comment:
This is 3 new lines for 6 saved in the test because now you can use `:=` there.

```suggestion
func generateTestCertWithValidity(t *testing.T, notBeforeStr, notAfterStr string) *x509.Certificate {
	t.Helper()
	if notBeforeStr == "" && notAfterStr == "" {
		return nil
	}
```